pkg/image/registry/imagestream/strategy.go

```diff
@@ -132,6 +142,37 @@ func (s Strategy) dockerImageRepository(stream *api.ImageStream) string {
 	return ref.String()
 }
 
+func (s Strategy) publicDockerImageRepository(stream *api.ImageStream) string {
+	// TODO: This can be more effective if we can use FieldSelector, however lister only
+	// support label selector. This also assumes that the route exists in the same namespace
+	// as the docker registry service.
+	for !s.routeInformer.Informer().HasSynced() {
```

Comment:
@smarterclayton the DefaultRegistry currently points to the service IP address, do we want something like "DefaultPublicRegistry" ? I don't think we want to change the DockerImageRepository to point to public URL as I think the builds should still push via internal service, right?